code/modules/mob/living/carbon/alien/special/facehugger.dm

```diff
@@ -1,265 +1,262 @@
-		return
-
-	if(iscarbon(target))
-		var/mob/living/carbon/C = target
-		if(C.wear_mask != src)
-			return
-
-	if(!sterile)
-		target.visible_message("<span class='danger'>[src] falls limp after violating [target]'s face!</span>", \
-								"<span class='userdanger'>[src] falls limp after violating your face!</span>")
-
-		Die()
-		icon_state = "[initial(icon_state)]_impregnated"
-
-		var/obj/item/bodypart/chest/LC = target.get_bodypart(BODY_ZONE_CHEST)
-		if((!LC || LC.status != BODYPART_ROBOTIC) && !target.getorgan(/obj/item/organ/body_egg/alien_embryo))
-			new /obj/item/organ/body_egg/alien_embryo(target)
-			var/turf/T = get_turf(target)
-			log_game("[key_name(target)] was impregnated by a facehugger at [loc_name(T)]")
-
-	else
-		target.visible_message("<span class='danger'>[src] violates [target]'s face!</span>", \
-								"<span class='userdanger'>[src] violates your face!</span>")
-
-/obj/item/clothing/mask/facehugger/proc/GoActive()
-	if(stat == DEAD || stat == CONSCIOUS)
-		return
-
-	stat = CONSCIOUS
-	icon_state = "[initial(icon_state)]"
-
-/obj/item/clothing/mask/facehugger/proc/GoIdle()
-	if(stat == DEAD || stat == UNCONSCIOUS)
-		return
-
-	stat = UNCONSCIOUS
-	icon_state = "[initial(icon_state)]_inactive"
+	//Check for mask
+	if(carbon_target.wear_mask)
+		var/obj/item/clothing/mask = carbon_target.wear_mask
+		if(istype(mask, /obj/item/clothing/mask/facehugger_item))
+			return FALSE
+		if(carbon_target.dropItemToGround(mask))
+			carbon_target.visible_message("<span class='danger'>[src] tears [mask] off of [target]'s face!</span>", "<span class='userdanger'>[src] tears [mask] off of your face!</span>")
+		else
+			carbon_target.visible_message("<span class='danger'>[src] tries to tear [mask] off of [target]'s face, but fails!</span>", "<span class='userdanger'>[src] trys to tear [mask] off of your face, but fails!</span>")
+			couple_timeout = world.time + couple_retry_time
+			return FALSE
 
-	addtimer(CALLBACK(src, .proc/GoActive), rand(MIN_ACTIVE_TIME, MAX_ACTIVE_TIME))
+	target.visible_message("<span class='danger'>[src] leaps at [target]'s face!</span>", "<span class='userdanger'>[src] leaps at your face!</span>")
+	var/obj/item/clothing/mask/facehugger_item/hugger_item = BecomeItem()
+	if(!carbon_target.wear_mask && carbon_target.equip_to_slot_if_possible(hugger_item, ITEM_SLOT_MASK, FALSE, TRUE, TRUE))
+		hugger_item.Attach(carbon_target)
+	return TRUE
+
+/**
+ * Turns the facehugger into an item, storing the mob inside of it.
+ *
+ * Proc to turn the facehugger into an item and store the mob inside said item.
+ * The item version will take the name, description, and appearance of the facehugger.
+ */
+/mob/living/simple_animal/hostile/facehugger/proc/BecomeItem()
+	var/obj/item/clothing/mask/facehugger_item/hugger_item = new(loc)
+	forceMove(hugger_item)
+	hugger_item.facehugger_mob = src
+	hugger_item.name = name
+	hugger_item.desc = desc
+	hugger_item.icon_state = icon_state
+	return hugger_item
```

Comment:
Why don't you hold the mask item inside the mob, same as you hold the mob inside the item?